CONTRIBUTING.md

```diff
@@ -0,0 +1,270 @@
+
+  1. Run the commands below:
+
+     ```
+     $ cd $GOPATH/src/github.com/opencontainers/runc
+     $ old_commit="..."
+     $ new_commit="..."
+     $ git log --no-merges --abbrev-commit --pretty=oneline "${old_commit}..${new_commit}") | sed 's/^/    /g'
+     ```
+
+  The output of the last command above should be pasted directly into the
+  commit "as-is". Note that the four space indent added by the `sed` command is
+  used to force github to render the list in a fixed-width font which makes the
+  list easier to read.
+
+## Patch format
+
+Beside the `Signed-off-by` footer, we expect each patch to comply with the
+following format:
+
+```
+Subsystem: Change summary (no longer than 75 characters)
+
+More detailed explanation of your changes: Why and how.
+Wrap it to 72 characters.
+
+See:
+    http://chris.beams.io/posts/git-commit/
+
+For some more good advice, and the Linux Kernel document:
+    https://git.kernel.org/cgit/linux/kernel/git/torvalds/linux.git/tree/Documentation/SubmittingPatches
+
+Fixes: #nnn
+
+Signed-off-by: <contributor@foo.com>
+```
+
+For example:
+
+```
+pod: Remove token from Cmd structure
+
+The token and pid data will be hold by the new Process structure and
+they are related to a container.
+
+Fixes: #123
+
+Signed-off-by: Sebastien Boeuf <sebastien.boeuf@intel.com>
+```
+
+Correct formatting of the PR patches is verified using the
+[checkcommits](https://github.com/kata-containers/tests/tree/master/cmd/checkcommits)
+tool.
+
+Note, that the body of the message should not just be a continuation of the
+subject line, and is not used to extend the subject line beyond its length
+limit. They should stand alone as complete sentence and paragraphs.
+
+It is recommended that each of your patches fixes one thing. Smaller patches
+are easier to review, and are thus more likely to be accepted and merged, and
+problems are more likely to be picked up during review.
+
+## Reviews
+
+Before your PRs are merged into the main code base, they will be
+reviewed. Anybody can review any PR and leave feedback (in fact, it
+is encouraged).
+
+We use an "acknowledge" system for people to note if they agree, or disagree,
+with a PR. We utilise some automated systems that can spot common
+acknowledge patterns, which include placing any of these **at the beginning of a
+comment line**:
+
+ - LGTM
+ - lgtm
+ - +1
+ - Approve
+
+### Examples
+
+This is an example of a valid "ack":
+
+```
+Excellent work - thanks for your contribution.
+
+lgtm
+```
+
+But the comment below is *not* valid because the magic "lgtm" does not start
+at the beginning of the line:
+
+```
+I love it! Very clean code and great tests. lgtm.
+```
+
+## Contact
+
+The Kata Containers community can be reached through its dedicated IRC
+channels and mailing lists:
+
+* IRC:
+  * Development discussions: `#kata-dev @ freenode.net`.
+  * General discussions: `#kata-general @ freenode.net`.
+
+* [Mailing lists](http://lists.katacontainers.io/).
+
```

Comment:
should we add a slack ref in here as well?

Reply:
Good idea. Adding...